python/build.py

```diff
@@ -27,11 +27,18 @@
     package['section'] = section_names[package.get('section', 'miscellaneous').lower()]
 
     print(f"  {package['repo']} -> {package['section']}")
-    try:
-        package['user'], package['repo_name'] = package['repo'].split('/')
-    except ValueError:
-        warnings.warn(f'Package.repo is not in correct format: {package}')
-        continue
+
+    if not package['repo'].startswith("http"):
+        try:
+            # try outdated GitHub user/repo_name format
+            _, package['repo_name'] = package['repo'].split('/')
+            # set proper repository URL
+            package['repo'] = f"https://github.com/{package['repo']}"
+        except ValueError:
+            warnings.warn(f'Package.repo is not in correct format: {package}')
+            continue
+
+    package.setdefault('repo_name', package['name'])
```

Comment:
I'm slightly worried about this. `name` is the name for display and may not be a legal reponame (e.g. "Farrow&Ball Matplotlib"). This is brittle. It works as long as we have a `repo_name` (i.e. the old GitHub user/repo_name format), but may fail for other `repo` urls if the do not specify `repo_name` explicitly.

`repo_name` is currently used for `conda_package`, `pypi_name`, `rtd_name` and `site`. Overall, I think we should move away from assumtions and towards simple explicit declarations. But that may be a bit much for this PR.

There are two ways forward:
- Get rid of all the inference and assumtions: This means somebody has to go through all the config files, and write down things explicitly.
- Only use `package['name']` if it's a reasonably simple identifier, e.g. (`re.match(r'^[\w-]+$', package['name'])`. But that means 'repo_name' is not always available and would need to be handled on the usage sites.

I'd be ok, to accept this PR as is if we open a follow-up issue to go down one these roads.